Bad code generation involving Modelica.Math.Nonlinear.solveOneNonlinearEquation

### Description

Try to simulate the following model
```Modelica
model TestMoistAirInverseVariables
  package Medium = Modelica.Media.Air.MoistAir;
  Medium.AbsolutePressure P;
  Medium.SpecificEnthalpy h;
  Medium.MassFraction Xi[Medium.nXi];
  Medium.Temperature T;
  Medium.ThermodynamicState state = Medium.setState_phX(P, h, Xi);
equation
  P = 98900;
  Xi[1] = 0.0007377581992833146;
  T = 20 + 273.15;
  state.T = T;
  annotation(uses(Modelica(version="4.0.0")));
end TestMoistAirInverseVariables;
```
Invalid C-code is generated
```C
TestMoistAirInverseVariables_02nls.c:40:184: error: use of undeclared identifier 'tmpMeta'
  res[0] = -293.15 + omc_Modelica_Math_Nonlinear_solveOneNonlinearEquation(threadData, (modelica_fnptr) mmc_mk_box2(0,closure0_TestMoistAirInverseVariables_Medium_T__phX_f__nonlinear,tmpMeta[0]), 190.0, 647.0, 1e-13);
```
there is no declaration of tmpMeta in the generated C code
```C
void residualFunc6(void** dataIn, const double* xloc, double* res, const int* iflag)
{
  TRACE_PUSH
  DATA *data = (DATA*) ((void**)dataIn[0]);
  threadData_t *threadData = (threadData_t*) ((void**)dataIn[1]);
  const int equationIndexes[2] = {1,6};
  int i;
  real_array tmp0;
  /* iteration variables */
  for (i=0; i<1; i++) {
    if (isinf(xloc[i]) || isnan(xloc[i])) {
      for (i=0; i<1; i++) {
        res[i] = NAN;
      }
      return;
    }
  }
  data->localData[0]->realVars[3] /* h variable */ = xloc[0];
  /* backup outputs */
  /* pre body */
  /* body */
  array_alloc_scalar_real_array(&tmp0, 1, (modelica_real)0.0007377581992833146);
  tmpMeta[0] = mmc_mk_box3(0, mmc_mk_real(98900.0), mmc_mk_real(data->localData[0]->realVars[3] /* h variable */), mmc_mk_modelica_array(tmp0));
  res[0] = -293.15 + omc_Modelica_Math_Nonlinear_solveOneNonlinearEquation(threadData, 
  (modelica_fnptr) mmc_mk_box2(0,closure0_TestMoistAirInverseVariables_Medium_T__phX_f__nonlinear,tmpMeta[0]), 190.0, 647.0, 1e-13);
  /* restore known outputs */
  TRACE_POP
}
```
### Expected Behavior

Correct C code should be generated

--- a/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
+++ b/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
@@ -4809,8 +4809,10 @@ template tempDecl(String ty, Text &varDecls)
     match ty /* TODO! FIXME! UGLY! UGLY! hack! */
       case "modelica_metatype"
       case "metamodelica_string"
-      case "metamodelica_string_const"
-        then 'tmpMeta[<%System.tmpTickIndex(1)%>]'
+      case "metamodelica_string_const" then
+        let newVarIx = 'tmpMeta<%System.tmpTick()%>'
+        let &varDecls += 'modelica_metatype <%newVarIx%>;<%\n%>'
+        newVarIx
       else
         let newVarIx = 'tmp<%System.tmpTick()%>'
         let &varDecls += '<%ty%> <%newVarIx%>;<%\n%>'
@@ -4834,8 +4836,10 @@ template tempDeclZero(String ty, Text &varDecls)
     match ty /* TODO! FIXME! UGLY! UGLY! hack! */
       case "modelica_metatype"
       case "metamodelica_string"
-      case "metamodelica_string_const"
-        then 'tmpMeta[<%System.tmpTickIndex(1)%>]'
+      case "metamodelica_string_const" then
+        let newVarIx = 'tmpMeta<%System.tmpTick()%>'
+        let &varDecls += 'modelica_metatype <%newVarIx%>;<%\n%>'
+        newVarIx
       else
         let newVarIx = 'tmp<%System.tmpTick()%>'
         let &varDecls += '<%ty%> <%newVarIx%> = 0;<%\n%>'

